tools/expotools/src/commands/UpdateReactNativeDocs.ts

```diff
@@ -0,0 +1,252 @@
+    if (typeof entry === 'string') {
+      return entry;
+    }
+  });
+
+  for (const entry of relevantDocs.flat()) {
+    const docExists = await fs.pathExists(path.join(docsPath, `${entry}.md`));
+    const docIsIgnored = DOCS_IGNORED.includes(entry);
+
+    if (docExists && !docIsIgnored) {
+      upstreamDocs.push(entry);
+    }
+  }
+
+  return upstreamDocs;
+}
+
+function getDocsSummary(localFiles: string[], upstreamFiles: string[]): DocsSummary {
+  const removed = localFiles.filter(entry => !upstreamFiles.includes(entry));
+  const added = upstreamFiles.filter(entry => !localFiles.includes(entry));
+
+  const changed = upstreamFiles.filter(entry => (
+    !(removed.includes(entry) || added.includes(entry))
+  ));
+
+  return { removed, added, changed };
+}
+
+async function applyRemovedFilesAsync(options: Options, summary: DocsSummary) {
+  if (!summary.removed.length) {
+    return console.log(`Upstream did not ${chalk.red('remove')} any files`);
+  }
+
+  for (const entry of summary.removed) {
+    if (entry.startsWith(PREFIX_REMOVED)) {
+      continue;
+    }
+
+    await fs.move(
+      path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${entry}.md`),
+      path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${PREFIX_REMOVED}${entry}.md`),
+    );
+  }
+
+  console.log(`Upstream ${chalk.red(`removed ${summary.removed.length} files`)}, see "${PREFIX_REMOVED}*.md" files.`);
+}
+
+async function applyAddedFilesAsync(options: Options, summary: DocsSummary) {
+  if (!summary.added.length) {
+    return console.log(`Upstream did not ${chalk.green('add')} any files`);
+  }
+
+  for (const entry of summary.added) {
+    if (entry.startsWith(PREFIX_ADDED)) {
+      continue;
+    }
+
+    await fs.copyFile(
+      path.join(RN_WEBSITE_DIR, 'docs', `${entry}.md`),
+      path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${PREFIX_ADDED}${entry}.md`),
+    );
+  }
+
+  console.log(`Upstream ${chalk.green(`added ${summary.removed.length} files`)}, see "${PREFIX_ADDED}*.md" files.`);
+}
+
+async function applyChangedFilesAsync(options: Options, summary: DocsSummary) {
+  if (!summary.changed.length) {
+    return console.log(`Upstream did not ${chalk.yellow('change')} any files`);
+  }
+
+  for (const entry of summary.changed) {
+    const upstreamDir = path.join(RN_WEBSITE_DIR, 'docs');
+    const diffPath = path.join(DOCS_DIR, 'pages', 'versions', options.sdk, 'react-native', `${entry}.diff`);
+
+    const { output: diff } = await spawnAsync('git', ['format-patch', `${options.from}..HEAD`, '--relative', `${entry}.md`, '--stdout'], { cwd: upstreamDir });
+    await fs.writeFile(diffPath, diff.join(''));
+  }
+
+  console.log(`Upstream ${chalk.yellow(`changed ${summary.changed.length} files`)}, see "*.diff" files.`);
+}
+
+interface Options {
+  sdk: string;
+  from: string;
+  to: string;
+}
+
+interface DocsSummary {
+  removed: string[];
+  added: string[];
+  changed: string[];
+}
```

Comment:
We usually put types/interfaces on top of the file. Also, they should probably be changed to types — as they are not intended to be implemented by classes 🙂